### DISCUSSION TITLE
Documentation: A Discovery Summary and Guide

### DISCUSSION THREAD

**Discussion by philleepflorence** [directus/directus#4326] [Feature Requests]
## What

This is a summary and guide for improving documentation as well as making documentation discoverable.

### What is Documentation

Documentation is any communicable material that is used to describe, explain or instruct regarding some attributes of an object, system, or procedure, such as its parts, assembly, installation, maintenance, and use.

### What is Software Documentation

Software documentation is written text or illustration that accompanies computer software or is embedded in the source code.

> **What is the purpose of Documentation**
> The documentation either explains how the software operates or how to use it. 
> _It can mean different things to people in different roles._

Content is considered discoverable when it is easily located either visually (when reading on a screen) or via query (searching via a search engine or internal search tool).

> **What is Discovery**
> The shortest distance between the source (resources and information) and the consumer (user) enabled by discovery learning.

### What is Discovery Learning

Discovery learning is a technique of inquiry-based learning, which is active learning that starts by posing questions, problems, or scenarios. Most learning online is some form or variation of discovery learning as there is no traditional teacher or guide presenting the information.

> How can discovery help
> A discoverable documentation can improve Product Adoption or Acceptance and User Retention.

## What are the Types of Software Documentation

This interpretation is focused on End User and Technical User Documentation - mostly common with open source projects.

#### Requirements
Statements that identify attributes, capabilities, characteristics, or qualities of a system.  
This is the foundation for what will be or has been implemented.

#### Architecture or Design
Overview of software.  
Includes relations to an environment and construction principles to be used in the design of software components.

#### Technical
Documentation of code, algorithms, interfaces, and APIs.  
This is essential to contributors to open source projects.

#### End User
Manuals for the end-user, system administrators, and support staff.
Anyone that would use the software.

#### Marketing   
How to market the product and analysis of the market demand.

## What are the Formats of User Documentation

User documentation can be produced or organized in a variety of online and print formats.

#### Tutorial   
A tutorial approach is considered the most useful for new users, in which they are guided through each step of accomplishing particular tasks.

#### Thematic.  
A thematic approach, where chapters or sections concentrate on one particular area of interest, is of more general use to an intermediate user. Some authors prefer to convey their ideas through a knowledge-based article to facilitate the user needs.   
This approach is usually practiced by a dynamic industry, where features are constantly added or the user population is largely correlated with the troubleshooting demands.

#### Reference. 
Reference or list is a format in which commands or tasks are simply listed alphabetically or logically grouped, often via cross-referenced indexes. This latter approach is of greater use to advanced users who know exactly what sort of information they are looking for.

#### End User
In product development, an end-user is a person who ultimately uses or is intended to ultimately use a product.  
These users are considered the ones that can make a product popular or not, depending on the ease of use of the product.

#### Technical User
In product development, a technical user is a person who ultimately can extend or is intended to ultimately extend a product primarily through the tools already available within the product.  

#### Contributor
A contributor is an advanced technical user that can improve the function of a product by contributing to the code.  
This is common in open-source projects.

## What are the Types of Content 

User documents don't need to be formatted in any particular way, but it is very important for them to have a thorough index, which is where the principles of discovery learning can help. 

There are four main types of content to convey information in documentation to a user.  While the names may intersect with the formats of user documentation, these focus on the content and the learning objectives.

> **How to Enable Discovery**
> To enable discovery, a **label** and **object** need to correspond to one learning objective - **Interrogative** and **Learning Object**. 

A **Learning Objective** relates to gaining an ability, a skill, some knowledge, what the user hopes to get after reading.
A **Label**, such as a heading or title is a variation of an **Interrogative** clause, whose form is typically associated with question-like meanings.
A **Learning Object** is a collection of content items, practice items, and assessment items that are combined based on a single learning objective.

#### What are the content discovery questions
Before creating any content, be sure to consider the following:

1. What is the learning objective of the content? 
2. Who are the users that need the content? 
3. How will the users locate the content?

### Reference 
A basic reference label and object **identify** and **define** a component or instruction.

1. **Interrogative** - what is it. what to do.
2. **Learning Objective** - Provide information and knowledge.
3. **Prerequisites** - Links to dependents, especially if dependent on other software.

### Concept
A basic concept label and object **describe** a component or task.
A concept can have a label and multiple objects if the objects extend a common idea, provided there are logical breaks, like paragraphs.

1. **Interrogative** - what does it do. what can it do.
2. **Learning Objective** - Provide understanding and comprehension.
3. **Prerequisites** - Reference.

### Guide
A basic guide label and object describe the method of using a component or performing a task.
A combination of guides can be together if they serve a common purpose.
The emphasis is on the concept or reference, but the guide can be discovered through the task by unfamiliar users (the function of the component as opposed to the name of the component).

1. **Interrogative** - how to use. how to extend.
2. **Learning Objective** - Enable problem-solving and application.
3. **Prerequisites**...Reference and Concept.

### Tutorial
A basic tutorial is a combination of different guides that describe a process not intrinsic to any one of the components.
It is common to link to concepts or references to keep the tutorial within the objective and to keep guides uniform.

1. **Interrogative** - how to do a task with. how to extend using.
2. **Learning Objective** - Enable and promote synthesis and creation.
3. **Prerequisites**... Reference, Concept, and Guides.

## Where
The primary sources for most of the information used.

[Documentation: A Discovery Interpretation - A Philleep Florence Blog](https://www.philleepflorence.com/-/blog/documentation-a-discovery-interpretation)

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
